src/Adapter/Supplier/QueryHandler/GetSupplierForViewingHandler.php

```diff
@@ -118,77 +117,116 @@ private function getSupplierProducts(Supplier $supplier, LanguageId $languageId)
-                        ];
-                    }
-                    $attribute = sprintf(
-                        '%s - %s',
-                        $combination['group_name'],
-                        $combination['attribute_name']
-                    );
-
-                    if (!empty($combinations[$attributeId]['attributes'])) {
-                        $attribute = sprintf(', %s', $attribute);
-                    }
-
-                    $combinations[$attributeId]['attributes'] = $attribute;
-                }
+        foreach ($supplierProducts as $productData) {
+            $product = new Product($productData['id_product'], false, $languageId->getValue());
+            $product->loadStockData();
+            $combinations = $this->findProductSupplierCombinations($product, $supplier, $languageId);
+
+            if (empty($combinations)) {
+                $products[] = $this->buildNonCombinationSupplierProduct($product, $supplier);
+                continue;
+            }
+
+            $products[] = [
+                'id' => $product->id,
+                'name' => $product->name,
+                'combinations' => $combinations,
+            ];
+        }
+
+        return $products;
+    }
+
+    /**
+     * @param Product $product
+     * @param Supplier $supplier
+     *
+     * @return array<string, mixed>
+     *
+     * @throws LocalizationException
+     */
+    private function buildNonCombinationSupplierProduct(Product $product, Supplier $supplier): array
+    {
+        $productInfo = Supplier::getProductInformationsBySupplier($supplier->id, $product->id);
+        $product->wholesale_price = $productInfo['product_supplier_price_te'];
+        $product->supplier_reference = $productInfo['product_supplier_reference'];
+        $isoCode = Currency::getIsoCodeById((int) $productInfo['id_currency']) ?: $this->defaultCurrencyIsoCode;
+        $formattedWholesalePrice = null !== $product->wholesale_price
+            ? $this->locale->formatPrice($product->wholesale_price, $isoCode)
+            : null;
+
+        return [
+            'id' => $product->id,
+            'name' => $product->name,
+            'reference' => $product->reference,
+            'supplier_reference' => $product->supplier_reference,
+            'wholesale_price' => $formattedWholesalePrice,
+            'ean13' => $product->ean13,
+            'upc' => $product->upc,
+            'quantity' => $product->quantity,
+            'combinations' => [],
+        ];
+    }
+
+    /**
+     * @param Product $product
+     * @param Supplier $supplier
+     * @param LanguageId $languageId
+     *
+     * @return array<int, array<string, mixed>>
+     *
+     * @throws LocalizationException
+     */
+    private function findProductSupplierCombinations(Product $product, Supplier $supplier, LanguageId $languageId): array
+    {
+        $productCombinations = $product->getAttributeCombinations($languageId->getValue());
+        if (!$productCombinations) {
```

Comment:
Product::getAttributeCombinations is supposed to always return an array

Reply:
That's what the PHPDoc says but when no combinations exist the SQL query can return false

Reply:
Seems like when there is no combinations an empty array is returned actually, so I replaced this check with `empty($productCombinations)`